src/test/java/com/thangle/domain/book/BookServiceTest.java

```diff
@@ -33,7 +38,135 @@ void shouldFindAll_OK() {
 
         verify(bookStore).findAll();
     }
+
+    @Test
+    void shouldFindByTitleAuthorDescription_OK() throws Exception {
+        final var book = buildBook();
+        final var expected = buildBooks();
+
+        when(bookStore.find(book.getTitle())).thenReturn(expected);
+
+        final var actual = bookService.find(book.getTitle());
+
+        assertEquals(expected, actual);
+
+        verify(bookStore).find(book.getTitle());
+    }
+
+    @Test
+    void shouldCreate_OK() {
+        final var book = buildBook();
+        when(bookStore.save(book)).thenReturn(book);
+
+        final var updatedBook = bookService.create(book);
+
+        assertEquals(book, updatedBook);
+
+        verify(bookStore).save(book);
+    }
+
+    @Test
+    void shouldCreate_WithTitleEmpty() {
+        final var book = buildBook();
+        book.setTitle("");
+
+        assertThrows(BadRequestException.class, () -> bookService.create(book));
+    }
+
+    @Test
+    void shouldCreate_WithAuthorEmpty() {
+        final var book = buildBook();
+        book.setAuthor("");
+
+        assertThrows(BadRequestException.class, () -> bookService.create(book));
+    }
+
+    @Test
+    void shouldCreate_WithUserIdNull() {
+        final var book = buildBook();
+        book.setUserId(null);
+
+        assertThrows(BadRequestException.class, () -> bookService.create(book));
+    }
+
+    @Test
+    void shouldUpdate_OK() {
+        final var book = buildBook();
+        final var updatedBook = buildUpdatedBook(book.getId());
+
+        when(bookStore.findById(book.getId())).thenReturn(Optional.of(book));
+        when(bookStore.save(book)).thenReturn(updatedBook);
+
+        final var actual = bookService.update(book.getId(), updatedBook);
+        assertEquals(updatedBook.getId(), actual.getId());
+        assertEquals(updatedBook.getTitle(), actual.getTitle());
+        assertEquals(updatedBook.getAuthor(), actual.getAuthor());
+        assertEquals(updatedBook.getDescription(), actual.getDescription());
+        assertEquals(updatedBook.getCreatedAt(), actual.getCreatedAt());
+        assertEquals(updatedBook.getUpdatedAt(), actual.getUpdatedAt());
+        assertEquals(updatedBook.getImage(), actual.getImage());
+        assertEquals(updatedBook.getUserId(), actual.getUserId());
+    }
+
+    @Test
+    void shouldUpdate_ThrowsNotFound() {
+        final var book = buildBook();
+        final var updatedBook = buildUpdatedBook(book.getId());
+
+        when(bookStore.findById(book.getId())).thenReturn(Optional.empty());
+
+        assertThrows(NotFoundException.class, () -> bookService.update(book.getId(), updatedBook));
+    }
+
+    @Test
+    void shouldUpdate_WithTitleEmpty() {
+        final var book = buildBook();
+        final var updatedBook = buildUpdatedBook(book.getId());
+
+        when(bookStore.findById(book.getId())).thenReturn(Optional.of(updatedBook));
+        updatedBook.setTitle("");
```

Comment:
Đưa lệnh này lên trước
```suggestion
        updatedBook.setTitle("");
```